### DISCUSSION TITLE
AI Guitar Amplifier Modelling - Welcome!

### DISCUSSION THREAD

**Discussion by Alec-Wright** [Alec-Wright/AI-GuitarAmpModelling#2] [General]
## 👋 Welcome!
I'm opening this discussions as a place for people to ask questions or leave comments or discuss anything in this repo, or about guitar amplifier modelling in general! Feel free to introduce yourself below!

**Comment by danielvachalek** [comment]
Hi @Alec-Wright, this is super interesting. As far as collecting training data, is it absolutely necessary to have two versions of the exact same riff, at the same speed and timing, and line up the two clips? How forgiving is the training/validation process if there is a slight mismatch or delay between the two clips?

**Reply by Alec-Wright** [reply]
Hi, it is generally necessary I would say. For the convolutional model it can handle the target being slightly delayed with respect to the input, but only if that delay is constant, this is just because the model can learn to apply a constant delay to the input signal, thus resynchronising the input/target. The way the training works kind of necessitates that the target dataset is actually the same underlaying audio, only processed by the guitar amplifier. This is basically because if you compare, for example, someone playing the exact same riff, twice, the two signals will still look very different to each other on a sample by sample basis if you compare them in the time-domain. So if your input/target signals are not from the same original recording, the model doesn't really have enough information to process the input signal and make it similar to the target.

**Comment by malegaultgit** [comment]
Hi, what a great project here, thank you  very much for your contribution!

I am in the process of trying to run the script myself, but unfortunately I am not able yet.. I would not say I am an advanced python programmer so sorry if my question might look basic..

I have installed Anaconda in Win10. I have download the repo an try to run "dist_model_recnet.py" from spider, without any argument. I am guessing running the script without any argument would run with default "demo" parameters. I have the following error: 

> runfile('C:/Automated-GuitarAmpModelling-main/dist_model_recnet.py', wdir='C:/Automated-GuitarAmpModelling-main')
> Reloaded modules: CoreAudioML, CoreAudioML.miscfuncs, CoreAudioML.training, CoreAudioML.dataset, CoreAudioML.networks
> no saved model found, creating new network
> Traceback (most recent call last):
> 
>   File "C:\Automated-GuitarAmpModelling-main\dist_model_recnet.py", line 137, in <module>
>     network = init_model(save_path, args)
> 
>   File "C:\Automated-GuitarAmpModelling-main\dist_model_recnet.py", line 105, in init_model
>     network.save_model('model', save_path)
> 
>   File "C:\Automated-GuitarAmpModelling-main\CoreAudioML\networks.py", line 54, in save_model
>     miscfuncs.dir_check(direc)
> 
>   File "C:\Automated-GuitarAmpModelling-main\CoreAudioML\miscfuncs.py", line 13, in dir_check
>     os.mkdir(dir_path)
> 
> FileNotFoundError: [WinError 3] The system cannot find the path specified: 'Results\\\jcm-RNN3'

I am guessing the code was initially run in an linux OS. Could there be an issue with the way path a handle with windows OS? If so, is there any small changes I should make so it runs smooth with my setup?

Thank you, Marc

**Reply by Alec-Wright** [reply]
The python 'os' package should handle the paths properly for Windows OS too. Do you have a directory called 'Results' in the same directory as the 'dist_model_recnet.py' file?

**Reply by malegaultgit** [reply]
Hi, thank you for your reply, everthing works now! (Of course) it was my fault, I did not download the files correctly.

I was able to run the code on two different computers, one without Cuda and running with CPU only, the other with a dedicated GPU and with Cuda. The later option took 8X less computing time, so it worth it (of course again!).

Thank you, Marc
